### PR TITLE
[WIP] Added agile template concepts to glossary and reviewed other concepts.

### LINKED ISSUE #366: Review Glossary
Oxford comma missing from a concept

### PR DESCRIPTION
**Pull request by Preeticp**
Adds agile concepts to glossary, addresses part of #382 and fixes #366

### CODE REVIEW

File: docs/topics/modules/glossary.adoc [hunk 1]
```diff
@@ -19,11 +19,12 @@ This file is consumed for the automatic generation of infotips used by OSIO web
   // endterm
 
 * When adding a new term, either use the `uuidgen` command on the CLI or use  https://www.uuidgenerator.net/ to generate a new UUID.
+Note: Planner UX provides uuids for each of the terms requiring infotips, use them rather than generating new ones.
```
Comment (kwk) [inline docs/topics/modules/glossary.adoc]: Some of the UUIDs come from the space template and UX might not know about those UUIDs upfront.

**Comment by Preeticp** [comment]
Received go ahead from Devs on the concepts. Going ahead and merging this with the understanding that this will only update the glossary in the User guide and not the Infotips on the UX.
Going forward, I will be raising a new PR for infotips in the agile.yaml file in the Planner repository.

@rkratky we might want to remove the UUIDs and the .json file but I suggest we do it at a later date, once we are assured that the new method suggested works well. Let me know your thoughts.

**Comment by kwk** [comment]
@Preeticp you should really setup your repo so only PRs with approval can be merged ;)

**Comment by Preeticp** [comment]
@kwk, do you have any modifications in mind for the definitions? This PR simply updates the concepts in the User Guide glossary. This PR and the draft for it has been circulating for almost the entire Sprint now. Comments in the draft were addressed. Also Michael mentioned I could go ahead with the definitions since Steve too raised no objections. I will be raising a separate PR for the infotips on Planner repo, let me know if you want me to revisit the definitions and revert this PR.

**Comment by Preeticp** [comment]
And yes, I have definitely missed a step here, @rkratky could you please review the infotips for Agile. Apologies for merging before peer review, I will raise a fresh PR to address any changes.

File: docs/topics/modules/glossary.adoc [hunk 2]
```diff
@@ -55,18 +56,39 @@ bug:: A defect that causes unexpected behavior in the software and that needs to
...
+epic:: A large user story, with a focused goal, that may need to be split into multiple user stories spanning over multiple iterations to be realized. It could comprise a short statement or phrase, a coherent aggregation of user stories and requirements, or an idea.
 // endterm
 
 ////
 // term: d0d33acf-7629-4653-9551-97ed9156f127, en
 (wig) epic:: A big chunk of work with a common objective that may take many iterations to deliver. It is split into multiple features and backlog items that try to satisfy the different aspects of the epic.
 // endterm
+
+// term: f450d7d0-3d38-4887-83ca-38dcd27c109b59, en
+(scrum)epic:: A big chunk of work with a common objective that may take many iterations to deliver. It is split into multiple features and backlog items that try to satisfy the different aspects of the epic.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: missing space after )

File: docs/topics/modules/glossary.adoc [hunk 3]
```diff
@@ -99,15 +121,14 @@ feature:: Features, often expressed as user stories, are required to actualize p
...
-impediment:: An impediment is anything that affects team performance or progress. It may include high probability risks that are likely to occur fairly soon.
+// term: 03b9bb64-4f65-4fa7-b165-494cd4f01401, en
+impediment:: Any issue or blocker that slows down the velocity of a team and affects team performance or progress.
 // endterm
-////
 
 ////
 // term: d5bd6f64-cee4-46d7-9b12-b3cb5e70145e, en
-issue:: An issue is an unmet user expectation, an observation that something doesn't work as one expects it to. It could represent bugs, feature requests, or enhancements and are linked to an appropriate work item in the backlog as part of the triage process. // endterm
+issue:: An issue is an unmet user expectation, an observation that something doesn't work as one expects it to. It could represent bugs, feature requests, or enhancements and are linked to an appropriate work item in the backlog as part of the triage process.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> does not
-> and is linked
-> a part of

File: docs/topics/modules/glossary.adoc [hunk 4]
```diff
@@ -120,6 +141,12 @@ iteration:: A development cycle used to organize, plan, and execute work items i
 // endterm
 ////
 
+////
+// term: 49d1a19f-02b4-4a10-a774-5723299f8944, en
+(agile) iteration:: A development cycle used to organize, plan, and execute work items in a certain order. It comprises a logical mix of work items slated to be executed within the time frame of the iteration.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: _...work items ... to be executed ..._? I don't think WIs are executed. Rather, they're worked on, completed, finished, etc.

File: docs/topics/modules/glossary.adoc [hunk 5]
```diff
@@ -150,12 +177,12 @@ scenario:: A scenario identifies a real-world user problem to resolve. Addressin
 
 ////
 // term: 679a563c-ac9b-4478-9f3e-4187f708dd30, en
-scenarios:: Work item type group that focuses on the planning oriented work item types such as scenario, fundamental and papercuts.
+scenarios:: Work item type group that focuses on the planning oriented work item types such as scenario, fundamental, and papercuts.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> item-type
-> planning-oriented
-> work-item types
-> , such as

File: docs/topics/modules/glossary.adoc [hunk 6]
```diff
@@ -150,12 +177,12 @@ scenario:: A scenario identifies a real-world user problem to resolve. Addressin
...
 // endterm
 ////
 
 // term: eb05f2b6-8a3c-4054-b28c-3eb1a47c125f, en
-scenario-driven development:: An agile development methodology focused on real-world problems, or scenarios, described in the language and from the viewpoint of the user. Scenarios deliver particular value propositions and are realized through experiences.
+scenario-driven development:: An agile development methodology focused on real-world problems or scenarios, described in the language and from the viewpoint of the user. Scenarios deliver particular value propositions and are realized through experiences.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> scenarios that is described

File: docs/topics/modules/glossary.adoc [hunk 7]
```diff
@@ -170,6 +197,10 @@ security issue:: OpenShift.io analyzes the CVEs of all your dependencies and fla
 space:: A space is the equivalent of a project. Each iteration and work item must be attached to a space, and a team of people can be attached to a space in various roles. By default, a space contains at least one area and one iteration.
 // endterm
 
+// term: 6ff83406-caa7-47a9-9200-4ca796be11bb, en
+story:: Incremental work or requirements, identified collaboratively by the team and the product owner, to deliver a product feature. Typically, they consist of a description expressed from the user’s perspective in the format - As a _user_, I want to achieve _objective_, so that this _purpose_ is met. It has an estimated size and an acceptance criteria to validate the work.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> in the following format:
-> an objective
-> and acceptance criteria

File: docs/topics/modules/glossary.adoc [hunk 8]
```diff
@@ -180,6 +211,20 @@ task:: Work assigned to various team members to implement a feature. They are ge
...
 ////
 
+////
+// term: 2853459d-60ef-4fbe-aaf4-eccb9f554b34, en
+(agile) task:: Work assigned to various team members to implement a feature. They are generally measured in units of 0.5 days, for example, four hours, eight hours, sixteen hours.
+// endterm
+////
+
+// term: 5182fc8c-b1d6-4c3d-83ca-6a3c781fa18a, en
+theme:: A group of user stories with a common underlying thread. They have a broad organizational level focus and are used to track related user stories.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> organization-level

File: docs/topics/modules/glossary.adoc [hunk 9]
```diff
@@ -180,6 +211,20 @@ task:: Work assigned to various team members to implement a feature. They are ge
...
+// endterm
+
+// term: 676c75df-bc94-4cd1-a219-dd4283d35a16, en
+time criticality:: It captures how business value of a work item decreases over time, higher values indicate greater time criticality. It indicates the criticality of meeting a deadline to derive maximum business value.
```
Comment (rkratky) [inline docs/topics/modules/glossary.adoc]: -> the business value
-> over time. Higher

**Comment by Preeticp** [comment]
Thanks a lot for the quick review @rkratky I will fix these in a new PR.

File: docs/topics/modules/glossary.adoc [hunk 10]
```diff
@@ -55,18 +56,39 @@ bug:: A defect that causes unexpected behavior in the software and that needs to
...
 
-// term: f450d7d0-3d38-4887-83ca-38d27c109b59, en
-epic:: A big chunk of work with a common objective that may take many iterations to deliver. It is split into multiple features and backlog items that try to satisfy the different aspects of the epic.
+// term: 04049d50-a1b9-42ac-89b5-0382088cb7b6, en
+component:: A distinct subsystem within the system that performs common functions.
+// endterm
+
+
+// term: fce0921f-ea70-4513-bb91-31d3aa8017f1, en
+defect:: An unacceptable variation between the actual behavior and the expected behavior of the software.
```
[the remark this answers is not in the document]
Comment (Preeticp) [reply docs/topics/modules/glossary.adoc]: Does it really sound that convoluted?! Will work on it :)

**Pull request merged**